src/components/Answer/index.tsx

```diff
@@ -7,30 +9,72 @@ import MultiInputs from 'components/MultiInputs';
 import Nps from 'components/Nps';
 import Rating from 'components/Rating';
 import TextArea from 'components/TextArea';
-import { Answer as AnswerType } from 'types/answer';
+import { Answer as AnswerType, instanceOfAnswer } from 'types/answer';
 import { DisplayType, Question, getDisplayTypeEnum } from 'types/question';
-import { AnswerRequest } from 'types/request/surveySubmitRequest';
+import { AnswerRequest, instanceOfAnswerRequest } from 'types/request/surveySubmitRequest';
 
 export const answerDataTestIds = {
   base: 'answer__base',
 };
 
 interface AnswerProps {
   question: Question;
+  onAnswerChanged: (answers: AnswerRequest[]) => void;
 }
-const Answer = ({ question }: AnswerProps): JSX.Element => {
+
+const Answer = ({ question, onAnswerChanged }: AnswerProps): JSX.Element => {
   const displayTypeEnum = getDisplayTypeEnum(question);
 
   const onValueChanged = (answer: number | AnswerType | AnswerRequest) => {
-    // TODO
-    console.log(answer);
+    if (typeof answer === 'number') {
+      onAnswerChanged([{ id: question.answers[answer - 1].id, answer: '' }]);
+    } else if (instanceOfAnswer(answer)) {
+      onAnswerChanged([{ id: answer.id, answer: '' }]);
+    } else if (instanceOfAnswerRequest(answer)) {
+      onAnswerChanged([answer]);
+    }
   };
 
   const onValuesChanged = (answers: AnswerType[] | AnswerRequest[]) => {
-    // TODO
-    console.log(answers);
+    if (!isEmpty(answers)) {
+      if (instanceOfAnswer(answers[0])) {
+        const answerRequests: AnswerRequest[] = answers.map((answer) => ({
+          id: answer.id,
+          answer: '',
+        }));
+
+        onAnswerChanged(answerRequests);
+      } else if (instanceOfAnswerRequest(answers[0])) {
+        onAnswerChanged(answers as AnswerRequest[]);
+      }
+    }
   };
 
+  useEffect(() => {
+    // Set default answers
+    switch (displayTypeEnum) {
```

Comment:
we can bring it out as a separate function instead of by-passing the eslint

Reply:
@nvminhtue `eslint` suggests me to bring them in if I extract them to a function 🤣 
<img width="759" alt="image" src="https://github.com/manh-t/react-survey/assets/60863885/fc63d206-6e22-45c4-965c-ca0f30f1bdf4">

Reply:
or you can use the `useCallback` with similar dependencies, can you try it

Reply:
@nvminhtue I read the documentation of `useCallback`, and they suggest removing the need for the function dependency. I updated code to follow this approach 😄 
![image](https://github.com/manh-t/react-survey/assets/60863885/5b757fc8-10ec-4cf2-b5a5-6e06b4fd72b9)

Reply:
I meant, you are by-passing the `exhaustive-deps` on eslint, does it help to minimize the dependencies if we move the switch block out of scope?

Reply:
@nvminhtue ah I got you. Even we use `useCallback`, the `exhaustive-deps` warning still displays. It doesn't minimize the dependencies 😓 
![image](https://github.com/manh-t/react-survey/assets/60863885/e12682c8-7007-44df-8d02-adbc2c2f4c9f)

Reply:
🤔 why don't we add those suggested dependencies @manh-t